deepchem/models/tensorgraph/layers.py

```diff
@@ -365,6 +376,8 @@ def shared(self, in_layers):
     return copy
 
   def _get_scope_name(self):
+    if tfe.in_eager_mode():
+      return None
```

Comment:
I'm a bit confused by this -- I imagine sharing variables in eager is much easier but how does it work?

Reply:
Currently it's not supported.  For the purposes of this PR (invoking layers directly), there's no need to share weights.  You just invoke the same layer again on different inputs.  That's how Tensorflow layers are designed to be used.  In the followup PR to get TensorGraph working in eager mode, I do want to try to support sharing variables.  I haven't figured out how I'll implement that yet.